atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/GetAsyncDelegate.java

```diff
@@ -18,19 +18,29 @@
 
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
 
+import com.palantir.atlasdb.futures.AtlasFutures;
 import com.palantir.atlasdb.keyvalue.api.Cell;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.transaction.api.Transaction;
-import com.palantir.common.base.Throwables;
 
 public class GetAsyncDelegate extends ForwardingTransaction {
 
     private final Transaction delegate;
+    private final Runnable enterFunction;
```

Comment:
so I suppose I'm also confused, we have a counter to make sure that we're tracking synchronous gets, are we running things in parallel such that we expect this to be > 1? I don't believe `get` is at any point recursive?

Reply:
I wouldn't be opposed to even having an `AtomicInteger` that is unused in the cases where you don't care about, since this lives in tests and I wouldn't be too concerned about perf.

Reply:
This is now resolved with an interface.